pkg/scheduler/framework/interface.go

```diff
@@ -608,6 +609,25 @@ type Handle interface {
 	Parallelizer() parallelize.Parallelizer
 }
 
+// PreFilterResult wraps needed info for scheduler framework to act upon PreFilter phase.
+type PreFilterResult struct {
+	// The set of nodes that should be considered downstream; if nil then all nodes are eligible.
+	NodeNames sets.String
+}
+
+func (p *PreFilterResult) Merge(in *PreFilterResult) {
+	if p == nil || in == nil {
+		return
+	}
+	if p.NodeNames == nil && len(in.NodeNames) != 0 {
+		p.NodeNames = sets.NewString()
+	}
+	for k := range in.NodeNames {
```

Comment:
I suppose they are intersected.

Reply:
Yeah. That's why we need to `Merge` them?